restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -1101,7 +1101,13 @@ private void onCallStateChanged(Object message, ActorRef sender) throws Transiti
                         if (dialBranches != null && dialBranches.contains(sender)) {
                             removeDialBranch(message, sender);
                         }
-                        checkDialBranch(message, sender, attribute);
+                        if (dialBranches != null && !dialBranches.isEmpty()){
```

Comment:
@ghjansen this shouldn't be removed. The intention here is the following:
1. Every BUSY branch will be removed from dialBranches
2. If dialBranches is empty, the checkDialBranch() method will ask for the next verb
3. If next verb is END, Bye will be sent to the call

If you cannot see the above, then maybe there is a race condition that moves the call to finishDialing, but there also the Call should get a Bye.